### ISSUE TITLE
Upgrade to analyzer 1.1.0

### ISSUE THREAD

**Issue by proninyaroslav** [vitusortner/floor#502]
`floor_generator 1.0.0-nullsafety.0` requires `analyzer ^0.40.6`.

**Comment by proninyaroslav** [comment]
I tried to `dependency_overrides: analyzer: ^ 1.1.0` and it doesn't seem to work as expected:
```
[WARNING] stderr: ../../.pub-cache/hosted/pub.dartlang.org/floor_generator-1.0.0-nullsafety.0/lib/processor/database_processor.dart:115:48: Error: Property 'element' cannot be accessed on 'DartType?' because it is potentially null.
[WARNING] stderr:  - 'DartType' is from 'package:analyzer/dart/element/type.dart' ('../../.pub-cache/hosted/pub.dartlang.org/analyzer-1.1.0/lib/dart/element/type.dart').
[WARNING] stderr: Try accessing using ?. instead.
[WARNING] stderr:         ?.map((object) => object.toTypeValue().element)
[WARNING] stderr:                                                ^^^^^^^
[WARNING] stderr: ../../.pub-cache/hosted/pub.dartlang.org/floor_generator-1.0.0-nullsafety.0/lib/processor/database_processor.dart:139:52: Error: Property 'element' cannot be accessed on 'DartType?' because it is potentially null.
[WARNING] stderr:  - 'DartType' is from 'package:analyzer/dart/element/type.dart' ('../../.pub-cache/hosted/pub.dartlang.org/analyzer-1.1.0/lib/dart/element/type.dart').
[WARNING] stderr: Try accessing using ?. instead.
[WARNING] stderr:             ?.map((object) => object.toTypeValue().element)
[WARNING] stderr:                                                    ^^^^^^^
[WARNING] stderr: ../../.pub-cache/hosted/pub.dartlang.org/floor_generator-1.0.0-nullsafety.0/lib/misc/extension/type_converter_element_extension.dart:19:50: Error: Property 'element' cannot be accessed on 'DartType?' because it is potentially null.
[WARNING] stderr:  - 'DartType' is from 'package:analyzer/dart/element/type.dart' ('../../.pub-cache/hosted/pub.dartlang.org/analyzer-1.1.0/lib/dart/element/type.dart').
[WARNING] stderr: Try accessing using ?. instead.
[WARNING] stderr:           ?.map((object) => object.toTypeValue().element);
[WARNING] stderr:                                                  ^^^^^^^
[WARNING] stderr: ../../.pub-cache/hosted/pub.dartlang.org/floor_generator-1.0.0-nullsafety.0/lib/misc/extension/type_converter_element_extension.dart:50:41: Error: Property 'element' cannot be accessed on 'InterfaceType?' because it is potentially null.
[WARNING] stderr:  - 'InterfaceType' is from 'package:analyzer/dart/element/type.dart' ('../../.pub-cache/hosted/pub.dartlang.org/analyzer-1.1.0/lib/dart/element/type.dart').
[WARNING] stderr: Try accessing using ?. instead.
[WARNING] stderr:   bool get isTypeConverter => supertype.element.displayName == 'TypeConverter';
[WARNING] stderr:                                         ^^^^^^^
[WARNING] stderr: ../../.pub-cache/hosted/pub.dartlang.org/floor_generator-1.0.0-nullsafety.0/lib/processor/entity_processor.dart:58:14: Error: Method 'toStringValue' cannot be called on 'DartObject?' because it is potentially null.
[WARNING] stderr:  - 'DartObject' is from 'package:analyzer/dart/constant/value.dart' ('../../.pub-cache/hosted/pub.dartlang.org/analyzer-1.1.0/lib/dart/constant/value.dart').
[WARNING] stderr: Try calling using ?. instead.
[WARNING] stderr:             .toStringValue() ??
[WARNING] stderr:              ^^^^^^^^^^^^^
[WARNING] stderr: ../../.pub-cache/hosted/pub.dartlang.org/floor_generator-1.0.0-nullsafety.0/lib/processor/entity_processor.dart:137:52: Error: The argument type 'List<String?>' can't be assigned to the parameter type 'List<String>' because 'String?' is nullable and 'String' isn't.
[WARNING] stderr:  - 'List' is from 'dart:core'.
[WARNING] stderr:             throw _processorError.noMatchingColumn(values);
[WARNING] stderr:                                                    ^
[WARNING] stderr: ../../.pub-cache/hosted/pub.dartlang.org/floor_generator-1.0.0-nullsafety.0/lib/processor/entity_processor.dart:163:23: Error: A value of type 'List<String?>' can't be returned from a function with return type 'List<String>' because 'String?' is nullable and 'String' isn't.
[WARNING] stderr:  - 'List' is from 'dart:core'.
[WARNING] stderr:             .toList() ??
[WARNING] stderr:                       ^
[WARNING] stderr: ../../.pub-cache/hosted/pub.dartlang.org/floor_generator-1.0.0-nullsafety.0/lib/processor/entity_processor.dart:219:14: Error: Method 'toBoolValue' cannot be called on 'DartObject?' because it is potentially null.
[WARNING] stderr:  - 'DartObject' is from 'package:analyzer/dart/constant/value.dart' ('../../.pub-cache/hosted/pub.dartlang.org/analyzer-1.1.0/lib/dart/constant/value.dart').
[WARNING] stderr: Try calling using ?. instead.
[WARNING] stderr:             .toBoolValue() ??
[WARNING] stderr:              ^^^^^^^^^^^
[WARNING] stderr: ../../.pub-cache/hosted/pub.dartlang.org/floor_generator-1.0.0-nullsafety.0/lib/processor/type_converter_processor.dart:21:51: Error: Property 'typeArguments' cannot be accessed on 'InterfaceType?' because it is potentially null.
[WARNING] stderr:  - 'InterfaceType' is from 'package:analyzer/dart/element/type.dart' ('../../.pub-cache/hosted/pub.dartlang.org/analyzer-1.1.0/lib/dart/element/type.dart').
[WARNING] stderr: Try accessing using ?. instead.
[WARNING] stderr:     final typeArguments = _classElement.supertype.typeArguments;
[WARNING] stderr:                                                   ^^^^^^^^^^^^^
[WARNING] stderr: ../../.pub-cache/hosted/pub.dartlang.org/floor_generator-1.0.0-nullsafety.0/lib/processor/insertion_method_processor.dart:85:10: Error: Method 'toEnumValueString' cannot be called on 'DartObject?' because it is potentially null.
[WARNING] stderr:  - 'DartObject' is from 'package:analyzer/dart/constant/value.dart' ('../../.pub-cache/hosted/pub.dartlang.org/analyzer-1.1.0/lib/dart/constant/value.dart').
[WARNING] stderr: Try calling using ?. instead.
[WARNING] stderr:         .toEnumValueString();
[WARNING] stderr:          ^^^^^^^^^^^^^^^^^
```

**Comment by ervindobri** [comment]
I support this, can't use floor with google_fonts for this reason.
Can we expect a fix soon?

**Comment by vitusortner** [comment]
We're looking into it right now and we'll create a new release ASAP.

**Comment by proninyaroslav** [comment]
Thank you, I hope for a quick release. Without this, I can't migrate my project to nullsafety.

**Comment by vitusortner** [comment]
We just published the 1.0.0-nullsafety.1 release. We unluckily can't go stable as crucial dependencies haven't yet been migrated. If you encounter problems with 1.0.0-nullsafety.1, please raise a new issue on GitHub.

**Comment by proninyaroslav** [comment]
Ok, thank you.
